diff -r f279f620339d -r 31e5f228a58d ic.cxx
--- a/ic.cxx	Thu Oct 13 20:43:54 2016 -0500
+++ b/ic.cxx	Fri Oct 14 13:39:04 2016 -0500
@@ -266,9 +266,16 @@
             const double age = param.ic.oceanic_plate_age_in_yr * YEAR2SEC;
             const MatProps &mat = *var.mat;
             const double diffusivity = mat.k(0) / mat.rho(0) / mat.cp(0); // thermal diffusivity of 0th element
+            double depth;
 
             for (int i=0; i<var.nnode; ++i) {
-                double w = -(*var.coord)[i][NDIMS-1] / std::sqrt(4 * diffusivity * age);
+
+              if (param.ic.is_topo_considered){
+                depth = get_actual_depth(var, (*var.coord)[i][NDIMS-1], (*var.coord)[i][0]);
+              }else{
+                depth = (*var.coord)[i][NDIMS-1];
+              }
+                double w = - depth/ std::sqrt(4 * diffusivity * age);
                 temperature[i] = param.bc.surface_temperature +
                     (param.bc.mantle_temperature - param.bc.surface_temperature) * std::erf(w);
             }
